contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,491 @@
+# Federated StatefulSet design
+irfan.rehman@huawei.com
+
+# Goal
+
+To outline the use cases for the need of federated Statefulset and determine
+suitable design/implementation approach for the same.
+
+# Background
+
+The original design proposal of the statefulset in kubernetes, including the
+use cases and example applications, can be found [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/stateful-apps.md).
+
+# Use Cases
+
+1 – A stateful app, wants replicas distributed in multiple clusters, such that it can
+form multiple smaller sized local app clusters using only local replicas, each serving
+users local to the cluster (using in cluster stateful app identities).
+
+2 – A stateful app, for the reasons of high availability, wants the stateful pods
+distributed in different clusters, such that the set can withstand cluster failures.
+This represents an app with one single global quorum. This use case can be an extension
+to the previous, where the same app instances get multiple identities (local and global).
+
+# Concrete use case examples
```

Comment:
Done! Partitioned into:
- examples for local identities only
- examples for global identities only
- examples needing multiple identities